.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+name: benchmarks
+
+on:
+  workflow_call:
+
+jobs:
+  benchmarks:
+    runs-on: ubuntu-latest
+    env:
+      COVALENT_SERVER_IFACE_ANY: true
+    steps:
+      - name: Check out head
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+
+      - name: Install Covalent
+        run: pip install -e .
+
+      - name: Install test requirements
+        run: |
+          pip install --no-cache-dir -r ./tests/requirements.txt
+          pip install --no-cache-dir jupyter
```

Comment:
ad hoc packages shouldn't be included here.  if this is for running notebooks, you should be using `nbconvert` which is already in the test requirements.